corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -270,6 +268,10 @@ hqDefine("data_dictionary/js/data_dictionary", [
                             'removeFHIRResourcePropertyPath': element.removeFHIRResourcePropertyPath(),
                             'allowed_values': pureAllowedValues,
                         };
+                        if (!element.hasChanges
+                            && data.index === element.index) {
+                            return;
+                        }
```

Comment:
Can we add this check above itself? It seems we are doing a lot of operations that we would throw away if we delay the check.

Reply:
That sounds like a good idea, I've moved the check in ([97026fc](https://github.com/dimagi/commcare-hq/pull/34781/commits/97026fcf7bf8320a77b63f4099261c8021427d66)).